include/gdsb/graph_input.h

```diff
@@ -325,6 +325,19 @@ std::tuple<Vertex64, uint64_t> read_binary_graph_partition(std::ifstream& input,
     return std::make_tuple(data.vertex_count, edge_count);
 }
 
+namespace binary
+{
+
+void read(std::ifstream&, gdsb::Edge32&);
+
+void read(std::ifstream&, gdsb::WeightedEdge32&);
+
+void read(std::ifstream&, gdsb::TimestampedEdge32&);
+
+void read(std::ifstream&, gdsb::WeightedTimestampedEdge32&);
+
```

Comment:
I think that's a matter of taste, but you could think about using read, read_weighted, read_timestamped and read_weighted_timestamped. I see that your option is cleaner for reading the code but maybe the explicit description of what the function does helps. You can think about it, its just a comment :)

Reply:
In this case I think that the parameter types clearly describe what we are reading: thus I document my code by the types.